app/controllers/tasks_controller.rb

```diff
@@ -7,6 +7,20 @@ def create
         @task = Task.create(task_params)
     end
 
+    def edit
+        @task = Task.find(params[:id])
+    end
+
+    def update
+        @task = Task.find(params[:id])
+        @task.update! (task_params)
+        redirect_to @task
+    end
+
+    def show
+        @task = Task.find(params[:id])
+    end
+
     private
 
```

Comment:
Refactor suggestions 😃  since `@task = Task.find(params[:id])` is repetitive

```suggestion
    before_action :set_task, only: %i[edit, update, show]
    def edit
    end

    def update
        @task.update! (task_params)
        redirect_to @task
    end

    def show
    end

    private
    def set_task
        @task = Task.find(params[:id])
    end
```